### PR TITLE
Show all files in file explorer and some extent of refactoring

### LINKED ISSUE #31902: VS not showing all files for all target OSs
Right not it is showing files for the first target, to see other OS files i need to switch position of target OSs.

@Anipik told me that VS should show all files for all targets

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/runtime/issues/31902

Currently we show files only for the first TargetFramework in TargetFrameworks list.
After this change
- we will be able to see all the files in the directory.
- the navigation, intelligence and refactoring will also work for these files.

. Renaming files won't work i.e wont update the references or the compile items in the csproj

### CODE REVIEW

File: src/libraries/Directory.Build.targets [hunk 1]
```diff
@@ -206,4 +206,9 @@
     <ExcludeFromPackage Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)' and '$(ExcludeCurrentNetCoreAppFromPackage)' == 'true'">true</ExcludeFromPackage>
     <ExcludeFromPackage Condition="'$(TargetFramework)' == '$(NetFrameworkCurrent)' and '$(ExcludeCurrentFullFrameworkFromPackage)' == 'true'">true</ExcludeFromPackage>
   </PropertyGroup>
+
+  <ItemGroup>
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets]: Please add a comment explaining why this is necessary

File: src/libraries/Directory.Build.targets [hunk 2]
```diff
@@ -206,4 +206,9 @@
...
+    <None Include="$(MSBuildProjectDirectory)\**\*.cs"
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets]:
```suggestion
    <None Include="$(MSBuildProjectDirectory)**\*.cs"
```

Comment (Anipik) [reply src/libraries/Directory.Build.targets]: The slash is required $(MSBuildProjectDirectory) doesnot have a trailing slash, and vs throws an error while trying to load this regex

Comment (ViktorHofer) [reply src/libraries/Directory.Build.targets]: > Do not include the final backslash on this property.

Oh right :(

**Review by ViktorHofer — APPROVED** [review]

File: src/libraries/Directory.Build.targets [hunk 3]
```diff
@@ -206,4 +206,11 @@
     <ExcludeFromPackage Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)' and '$(ExcludeCurrentNetCoreAppFromPackage)' == 'true'">true</ExcludeFromPackage>
     <ExcludeFromPackage Condition="'$(TargetFramework)' == '$(NetFrameworkCurrent)' and '$(ExcludeCurrentFullFrameworkFromPackage)' == 'true'">true</ExcludeFromPackage>
   </PropertyGroup>
+
+  <!-- The Default behavior in the VS is to show the files for first target framework in target frameworks property.
```
Comment (safern) [inline src/libraries/Directory.Build.targets]: the VS -> `VS`
```suggestion
  <!-- The Default behavior in VS is to show the files for first target framework in TargetFrameworks property.
```

File: src/libraries/Directory.Build.targets [hunk 4]
```diff
@@ -206,4 +206,11 @@
...
+        This is required to show all the files corresponding to all target frameworks in the VS -->
```
Comment (safern) [inline src/libraries/Directory.Build.targets]:
```suggestion
        This is required to show all the files corresponding to all target frameworks in VS -->
```

File: src/libraries/Directory.Build.targets (hunk 2 above)
Comment (safern) [reply src/libraries/Directory.Build.targets]: I guess this won't cause trouble by including duplicates with the ones included in the Compile item?

Comment (ViktorHofer) [reply src/libraries/Directory.Build.targets]: Those are excluded in the Exclude filter.

Comment (safern) [reply src/libraries/Directory.Build.targets]: Ah, I didn't see them as I just replied the question here and it didn't show all the lines 😄

File: src/libraries/Directory.Build.targets [hunk 5]
```diff
@@ -206,4 +206,11 @@
     <ExcludeFromPackage Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)' and '$(ExcludeCurrentNetCoreAppFromPackage)' == 'true'">true</ExcludeFromPackage>
     <ExcludeFromPackage Condition="'$(TargetFramework)' == '$(NetFrameworkCurrent)' and '$(ExcludeCurrentFullFrameworkFromPackage)' == 'true'">true</ExcludeFromPackage>
   </PropertyGroup>
+
+  <!-- The Default behavior in VS is to show the files for first target framework in TargetFrameworks property.
```
Comment (safern) [inline src/libraries/Directory.Build.targets]: One more, sorry.

```suggestion
  <!-- The Default behavior in VS is to show files for the first target framework in TargetFrameworks property.
```

Comment (Anipik) [reply src/libraries/Directory.Build.targets]: done

**Review by stephentoub — APPROVED** [review]

**Pull request merged**